src/java.base/share/classes/java/lang/invoke/MethodType.java

```diff
@@ -398,6 +400,13 @@ private static MethodType makeImpl(Class<?> rtype, Class<?>[] ptypes, boolean tr
             ptypes = NO_PTYPES; trusted = true;
         }
         MethodType primordialMT = new MethodType(rtype, ptypes);
+        if (AOTHolder.archivedMethodTypes != null) {
+            MethodType mt = AOTHolder.archivedMethodTypes.get(primordialMT);
```

Comment:
Can we populate these AOT MTs into the internTable so we only look up once?

Reply:
There's a trade off here -- start-up will be slowed down if the archivedMethodTypes table is big. Also, `internTable` uses more memory as it wraps every entry inside a weak reference.